pkg/inspecttypes/dockercompat/dockercompat.go

```diff
@@ -176,6 +242,45 @@ func ContainerFromNative(n *native.Container) (*Container, error) {
 	return c, nil
 }
 
+func ImageFromNative(n *native.Image) (*Image, error) {
+	i := &Image{}
+
+	imgoci := n.ImageSpec
+
+	i.RootFS.Type = imgoci.RootFS.Type
+	diffIDs := imgoci.RootFS.DiffIDs
+	for _, d := range diffIDs {
+		i.RootFS.Layers = append(i.RootFS.Layers, d.String())
+	}
+	i.Comment = imgoci.History[len(imgoci.History)-1].Comment
+	i.Created = imgoci.History[len(imgoci.History)-1].Created.String()
+	i.Author = imgoci.History[len(imgoci.History)-1].Author
+	i.Architecture = imgoci.Architecture
+	i.Os = imgoci.OS
+
+	portSet := make(nat.PortSet)
+	for k := range imgoci.Config.ExposedPorts {
+		portSet[nat.Port(k)] = struct{}{}
+	}
+
+	i.Config = &Config{
+		Cmd:          imgoci.Config.Cmd,
+		Volumes:      imgoci.Config.Volumes,
+		Env:          imgoci.Config.Env,
+		User:         imgoci.Config.User,
+		WorkingDir:   imgoci.Config.WorkingDir,
+		Entrypoint:   imgoci.Config.Entrypoint,
+		Labels:       imgoci.Config.Labels,
+		ExposedPorts: portSet,
+	}
+
+	i.ID = n.Descriptor.Digest.String()
+
+	s := []string{n.Image.Name, "@", n.Image.Target.Digest.String()}
+	i.RepoTags, i.RepoDigests = parseImageReferences([]string{strings.Join(s, "")})
```

Comment:
We already know the digest, no need to "parse" it

Reply:
I let you check :)